File: tests/class/copy-and-edit.swift
public class Point {
	var x: Double
	var y: Double

	init(x _x: Double, y _y: Double) {
		x = _x
		y = _y
	}
}

let origin = Point(x: 0, y: 0)

public func pointOffsetFromOrigin(x: Double, y: Double) -> Point {
	let result = Point(x: origin.x, y: origin.y)
	result.x += x
	result.y += y
	return result
}

File: tests/class/copy-and-edit.js
export function pointOffsetFromOrigin$x$y$(x, y) {
  const _x = origin.x;
  const _y = origin.y;
  const point0 = new Point();
  point0.x = _x;
  point0.y = _y;
  const result = point0;
  result.x = result.x + x;
  result.y = result.y + y;
  return result;
}
export class Point {}
const point = new Point();
point.x = 0;
point.y = 0;
const origin = point;